mlflow/pytorch/__init__.py

```diff
@@ -516,7 +536,37 @@ def _load_model(path, **kwargs):
             return torch.jit.load(model_path)
 
 
-def load_model(model_uri, **kwargs):
+def load_state_dict(model_uri, model_class_obj):
+    """
+    Load a PyTorch model state dict from a local file or a run.
+
+    :param model_uri: The location, in URI format, of the MLflow model, for example:
+
+                    - ``/Users/me/path/to/local/model``
+                    - ``relative/path/to/local/model``
+                    - ``s3://my_bucket/path/to/model``
+                    - ``runs:/<mlflow_run_id>/run-relative/path/to/model``
+                    - ``models:/<model_name>/<model_version>``
+                    - ``models:/<model_name>/<stage>``
+
+                    For more information about supported URI schemes, see
+                    `Referencing Artifacts <https://www.mlflow.org/docs/latest/concepts.html#
+                    artifact-locations>`_.
+    :param model_class_obj: Instance of the model class
+
+    :return: A pytorch model instance
+    """
+    import torch
+
+    model_path = load_model(model_uri, load_state_dict=True)
+    model_path = os.path.join(model_path, _SERIALIZED_TORCH_MODEL_FILE_NAME)
+
+    state_dict = torch.load(model_path)
+    model_class_obj.load_state_dict(state_dict)
+    return model_class_obj
+
+
+def load_model(model_uri, load_state_dict=False, **kwargs):
```

Comment:
Can we avoid introducing `load_state_dict` here and just provide `load_state_dict` instead?

Reply:
remove the boolean parameter. Refactored the existing `load_model` method to re-use most of the code in `load_state_dict`